tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -119,17 +119,15 @@ func PackageMetadataCmd() *cobra.Command {
 				mainSpec.Repository = "https://github.com/" + repoSlug.String()
 			default:
 				r, err := url.Parse(mainSpec.Repository)
-				if err == nil {
-					err = repoSlug.Set(r.Path)
-				}
 				if err != nil {
 					return errors.Wrapf(err, "parsing repo url %q from schema", mainSpec.Repository)
 				}
+
 				if r.Hostname() != "github.com" {
 					return fmt.Errorf("mainSpec.Repository host must be from GH, found %q", r.Hostname())
 				}
 
-				if err := repoSlug.Set(r.Path); err != nil {
+				if err := repoSlug.Set(strings.Trim(r.Path, "/")); err != nil {
 					return errors.Wrapf(err, "parsing repo slug from %q", mainSpec.Repository)
```

Comment:
`repoSlug.Set` is what parses the repo slug. The error is still correct.

It's `func (*repoSlug) Set` instead of `parseRepoSlug` to fulfill an interface.

Reply:
That feels like incorrect naming to me for the interface then.

I would expect the error to be "error setting the optional input foo", not "error parsing the input". 

This feels out of scope for this PR since it doesn't directly concern your changes, or the PR's intent, so I don't want to belabor it further, but it'd be great to clean up in the future.

Reply:
It's an interface defined in `cobra`. I agree that `Set` is an extremely broad word for "parse a string into a custom type for cobra", but its what they use.